Lesson4/src/Homework4/Services/MilitaryPart.java

```diff
@@ -0,0 +1,106 @@
+package Homework4.Services;
+
+import Homework4.Model.Person;
+
+import java.util.List;
+import java.util.Scanner;
+
+public class MilitaryPart {
+
+
+    private int size;
+    private Person[] sizeMilitary;
+
+
+    public MilitaryPart(List<Person> fitPeople) {
+        Scanner scanner = new Scanner(System.in);
+        System.out.print("Can you enter size Military part, please\t ");
+        size = scanner.nextInt();
+        sizeMilitary = new Person[size];
+        for (int i = 0; i < size; i++) {
+            sizeMilitary[i] = new Person(0, "UNNAMED", 0, "men");
+        }
+        int longList = 0;
+        for (Person i : fitPeople) {
+            longList++;
+        }
+        for (int i = 0; i < size; i++) {
+            if (i > longList - 1) {
+                break;
+            } else if (sizeMilitary[i].getName() == "UNNAMED") {
```

Comment:
you don't need to do this part

```
 for (int i = 0; i < size; i++) {
            sizeMilitary[i] = new Person(0, "UNNAMED", 0, "men");
}
```
because you can do verification like if (sizeMilitary[i] == null) and it will mean that sizeMilitary[i] is not taken. Same for if check in the newSoldier method